src/node_options.cc

```diff
@@ -575,6 +575,12 @@ EnvironmentOptionsParser::EnvironmentOptionsParser() {
             "write warnings to file instead of stderr",
             &EnvironmentOptions::redirect_warnings,
             kAllowedInEnvvar);
+  AddOption(
+      "[has_env_file_string]", "", &EnvironmentOptions::has_env_file_string);
+  AddOption("--env-file",
+            "set environment variables from supplied file",
+            &EnvironmentOptions::env_file);
```

Comment:
@anonrig Was it a deliberate decision to not set `kAllowedInEnvvar` for this option?

I'm asking, because in order to replace the `dotenv`-npm-package with this feature AND using npm scripts, I need to set `NODE_OPTIONS='--env-file .env'` in my `.npmrc` file.

Currently, this is not possible because `kAllowedInEnvvar` is not set.

Reply:
It wasn't a deliberate decision. Prs are welcome :-)

Reply:
Is there a risk of a loop here? Like what if an `.env` file contains `NODE_OPTIONS='--env-file .env'`?

Reply:
No there isn't

Reply:
Thanks for the info! I'll read through the contribution guidelines and open a PR.

Reply:
Sadly, setting `kAllowedInEnvvar` won't allow to use `NODE_OPTIONS='--env-file .env'`, since `NODE_OPTIONS` is parsed **after** `Dotenv` reads the command line arguments.

So one would have to adapt the `GetPathFromArgs`-function in order to also check if `--env-file` exists in `NODE_OPTIONS`.

I tried to implement a `Dotenv::GetPathFromArgsAndNodeOptions`-function, but my C knowledge is pretty bad, so I'm hoping that someone else will implement it in the future (since it would be really useful when running node-scripts via npm).

Reply:
> it would be really useful when running node-scripts via npm

How so? What is a script you would use it for?

Reply:
> How so? What is a script you would use it for?

In my case I'm running playwright via npm like this:

#### package.json

```json

{
	"name": "npm-package",
	"scripts": {
		"test:browser": "playwright test"
	}
}
```

Inside my playwright config, I'm currently using `dotenv.config()` to get the baseUrl for the current env.

#### playwright.config.ts
```js
import dotenv from 'dotenv';

dotenv.config();

const config: PlaywrightTestConfig = {
	baseURL: process.env.PLAYWRIGHT_BASE_URL,
};

export default config;
```

I'd like to get rid of the npm dotenv-dependency and use the native Dotenv functionality, without having to change my npm scripts to something like `node --env-file=.env node_modules/.bin/playwright test`.

Reply:
You can playwright with something like this: `node --env-file=.env ./node_modules/.bin/playwright test`

Reply:
> without having to change my npm scripts to something like `node --env-file=.env node_modules/.bin/playwright test`

So you're saying you prefer to do `NODE_OPTIONS --env-file=.env playwright test`? Node doesn't automatically load .env files.

Reply:
> So you're saying you prefer to do `NODE_OPTIONS=--env-file=.env playwright test`? Node doesn't automatically load .env files.

Yes, sort of.

As far as i know, I can use [`.npmrc`](https://docs.npmjs.com/cli/v10/configuring-npm/npmrc) and [`node-options`](https://docs.npmjs.com/cli/v10/using-npm/config#node-options) to automatically apply `NODE_OPTIONS` to called lifecycle scripts (like playwright).

Reply:
> As far as i know, I can use [`.npmrc`](https://docs.npmjs.com/cli/v10/configuring-npm/npmrc) and [`node-options`](https://docs.npmjs.com/cli/v10/using-npm/config#node-options) to automatically apply `NODE_OPTIONS` to called lifecycle scripts (like playwright).

Interesting, I didn’t know about [node-options](https://docs.npmjs.com/cli/v10/using-npm/config#node-options)!